### PR TITLE
Added a subclass of astropy.time to support utime

### PR DESCRIPTION
**Pull request by jmason86**
Implemented handling of utime by creating a custom subclass of ```astropy.time.Time```, which uses the [third code example from their documentation](http://docs.astropy.org/en/stable/time/#writing-a-custom-format) as a template.

### CODE REVIEW

**Comment by wafels** [comment]
Will need a test too.

**Comment by wafels** [comment]
Also, maybe a short explanation of why this is needed.

**Comment by jmason86** [comment]
Will do. It's needed because @Cadair told me so? 😛

**Comment by Cadair** [comment]
Knowing very little about this, can you construct it with the float? A test for that use would be great as well. (Seeing how that's the most likely use of this, not to save *out* utime as we are not monsters :p )

**Comment by jmason86** [comment]
I was afraid of you asking for that.. that's what I spent a couple hours on yesterday (but for a new yyyydoy_sod subclass) and that is much harder to figure out since I haven't found any comprehensive astropy examples. I'll keep at it though.

**Comment by jmason86** [comment]
This PR is now dependent on the resolution of [issue 7092 in astropy](https://github.com/astropy/astropy/issues/7092) that will enable, in this case, utime as an input.

File: sunpy/time/utime.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from __future__ import absolute_import
+
+from astropy.time.formats import erfa, TimeFromEpoch
+
+
+class TimeUTime(TimeFromEpoch):
+    """Seconds from 1979-01-01 00:00:00 UTC. Same as Unix time
```
Comment (nabobalis) [inline sunpy/time/utime.py]: I forget, should there not be a new line after the docstring quotes?

File: sunpy/time/utime.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    but this starts 9 years later.
+
+    This time format is included for historical reasons. Some
+    people in solar physics prefer using this epoch.
+
+    Example:
+        from astropy.time import Time
+        t = Time('2000-01-01T13:53:23')
+        print(t.utime)
+        >>> 662738003.0
```
Comment (nabobalis) [inline sunpy/time/utime.py]: This looks wrong for example syntax. 

I think it should be somrthing like:
```
Examples
--------
     >>> code line 1
     >>> code line 2
```
I forget how you add the output.

File: sunpy/time/utime.py (hunk 1 above)
Comment (jmason86) [reply sunpy/time/utime.py]: [PEP257](https://www.python.org/dev/peps/pep-0257/#id17) doesn't specify but I can change it if that's a standard recommended in sunpy or in the science stack.

File: sunpy/time/utime.py (hunk 2 above)
Comment (jmason86) [reply sunpy/time/utime.py]: I think you're right. I couldn't find a PEP standard but I found a [random example of how it's done in scipy](https://github.com/scipy/scipy/blob/master/scipy/interpolate/_bsplines.py). For example: 
```
 >>> from scipy.interpolate import BSpline
 >>> k = 2
 >>> t = [0, 1, 2, 3, 4, 5, 6]
 >>> c = [-1, 2, 0, -1]
 >>> spl = BSpline(t, c, k)
 >>> spl(2.5)
 array(1.375)
 >>> bspline(2.5, t, c, k)
 1.375
```

File: sunpy/time/utime.py (hunk 1 above)
Comment (nabobalis) [reply sunpy/time/utime.py]: It's not an issue then!

File: sunpy/time/utime.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+
+class TimeUTime(TimeFromEpoch):
+    """Seconds from 1979-01-01 00:00:00 UTC. Same as Unix time
+    but this starts 9 years later.
+
+    This time format is included for historical reasons. Some
+    people in solar physics prefer using this epoch.
+
+    Example:
```
Comment (nabobalis) [inline sunpy/time/utime.py]: I think example should be underlined

```
Example
-------
```

File: sunpy/time/utime.py (hunk 1 above)
Comment (Cadair) [reply sunpy/time/utime.py]: It's a Cadair standard, which means most of SunPy is like that ;)

File: sunpy/time/tests/test_utime.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+from astropy.time import Time
+
+
+def test_utime_t0():
+    assert Time('1979-01-01T00:00:00').utime == 0.0
+
+
+def test_utime_random_date():
+    assert Time('2018-01-13T13:32:56').utime == 1231853576.0
```
Comment (Cadair) [inline sunpy/time/tests/test_utime.py]: Could you please add a test which is marked with `@pytest.mark.xfail` of the reverse conversion? i.e. the open astropy issue?

File: sunpy/time/tests/test_utime.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+from __future__ import absolute_import, division, print_function
+
+from astropy.time import Time
+
```
Comment (Cadair) [inline sunpy/time/tests/test_utime.py]: should this not explicitly import the utime format to register it with astropy?

**Comment by Cadair** [comment]
@jmason86 can you merge master into this to fix the CI please?

**Comment by Cadair** [comment]
Can you add a changelog entry for this please? Other than that this looks great.

**Pull request merged**